src/Illuminate/Container/Container.php

```diff
@@ -1024,6 +1026,17 @@ protected function resolvePrimitive(ReflectionParameter $parameter)
             return [];
         }
 
+        if (isset($this['config']) &&
```

Comment:
untested code, but i think you could improve readability like this

```
if ($this->has('config') && is_subclass_of($this['config'], \Illuminate\Contracts\Config\Repository::class))
```

`is_subclass_of` will always return `false` for non existing classes/interfaces.

Reply:
Okay I will implement this !

Reply:
Test seems broken now, not sure if it's related though

Reply:
> Test seems broken now, not sure if it's related though

Seems to be unrelated, at least "tests / PHP 8.1 - prefer-lowest (pull_request)" is a deprecation warning in an unrelated part of Laravel.

Reply:
Has been fixed now